### ISSUE TITLE
Object spread should not keep non-enumerable properties

### ISSUE THREAD

**Issue by Touffy** [microsoft/TypeScript#43859]
# Bug Report

### 🔎 Search Terms

spread symbol

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAcuAomArgWxgXhgZQJ5oBGIANgBQCUA3AFA2iSwiEBWAgljAN4CGAXDABMAGhgBtBGGToAugIDMAX3rhoMZiwBCnLjAB0BjR2UM1GgMICoeAA4BTEADN1rbdl4DByoA)

### 💻 Code

```ts
const NonEnum = Symbol();

const objA = {a: 2, [NonEnum]: 3}
const objB = { ...objA }
const objC: typeof objB = {a: 2}
```

### 🙁 Actual behavior

objB is inferred as `{a: number, [NonEnum]: number}`, when it should have been just `{a: number}`

Consequently, the assignment to objC is reported as an error.

### 🙂 Expected behavior

objB should have been inferred as just `{a: number}` because object spread only spreads enumerable properties, and Symbols are never enumerable.

When spreading an instance of a class, class methods are correctly not copied to the result. This should be the same.

**Comment by nmain** [comment]
> Symbols are never enumerable

That's not correct.  Try your code in a browser console and see what happens.  Spread copies own enumerables, whether they be symbol or string, and symbol properties can be enumerable, and are in this case.

```js
console.log(objB[NonEnum]) // prints '3'
```

**Comment by Touffy** [comment]
Sorry, you're right. Closing this issue.
